test/boost/filtering_test.cc

```diff
@@ -1176,3 +1176,155 @@ SEASTAR_TEST_CASE(test_filtering) {
 
     });
 }
+
+SEASTAR_TEST_CASE(test_filtering_gt_not_capturing_nulls) {
```

Comment:
I would have liked to have a comment explaining in words (not C++ code) what this code is checking (maybe also referencing the issue), and give some justification why the tested behavor is really correct: Is it because of common sense? Because Cassandra behaves this way? Because SQL behaves this way? Or what?

Reply:
This really is a test of `null` on the LHS.  We already have a test named null (line 263) that tests `null` on the RHS.  It would be nice if both tests were renamed to emphasize this.

Reply:
- I added explanation
- I renamed both tests to `null_{l,r}hs` and put them into the same file